File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
n = int(input())
i = 1; j = 1
sum = 1
cnt = 1
while j != n:
  if sum == n:
    cnt += 1
    j += 1
    sum += j
  elif sum < n:
    j += 1
    sum += j
  else:
    sum -= i
    i += 1

print(cnt)

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

       int i = 1, j = 1, sum = 1, res = 0;
       while (i <= j) {
           if(n == sum) {
               res++;
               j++;
               sum += j;
           }
           else if(n < sum) {
               sum -= i;
               i++;
           }
           else {
               j++;
               sum += j;
           }
       }
        System.out.println(res);
    }
}